### DISCUSSION TITLE
AdguardHome cpu usage high during authentication

### DISCUSSION THREAD

**Discussion by yongman** [AdguardTeam/AdGuardHome#6562] [General]
I found AdguardHome cost a very high cpu usage peridically. Then I catch the cpu profile using pprof and it shows the most time is doing the authentication.

<img width="1405" alt="image" src="https://github.com/AdguardTeam/AdGuardHome/assets/3875656/0a7fba9b-da50-4e1e-bdff-1ead34974240">

The reason is that I connect adguard home to homeassistant for better view and control, but I want to figure out the rationality of the cpu cost in `findUser` being too high.

**Comment by ainar-g** [comment]
[Bcrypt](https://en.wikipedia.org/wiki/Bcrypt) to hash passwords, so if HomeAssistant is using basic-auth requests a lot, it will use bcrypt a lot.  An alternative would be for the developers of that integration to use the usual login API and use a session cookie instead.
